site/profile/manifests/it/graylog.pp

```diff
@@ -0,0 +1,169 @@
+# Graylog config
+class profile::it::graylog {
+  class { 'java' :
+  package => 'java-1.8.0-openjdk',
+    }
+# I don't why but this exact config retured error initially.
+# I had to specify version first with manage_package_repo to true
+# Then change things back to original and it installed.    
+class {'mongodb::globals':
+  manage_package_repo => true,
+  manage_package      => true,
+  version             => '3.6.18-1.el7',
+  }
+class { 'mongodb::server':
+  bind_ip => ['127.0.0.1'],
+  }
+$xms = lookup('elasticsearch_xms')
+$xmx = lookup('elasticsearch_xmx')
+
+class { 'elastic_stack::repo':
+  version => 6,
+  }
+
+class { 'elasticsearch':
+  version       => '6.3.1',
+  manage_repo   => true,
+    jvm_options => [
+      "-Xms${xms}",
+      "-Xmx${xmx}"
+    ]
+  }
+elasticsearch::instance { 'graylog':
+  config => {
+    'cluster.name' => 'graylog',
+    'network.host' => '127.0.0.1',
+    }
+  }
+class { 'graylog::repository':
+  version => '3.0'
+}
+#################################################################################################################
+# Requirements for graylog::server
+$ssl_config_dir = '/etc/graylog/server'
+
+#TODO fine a better way of resolve this dependency
+  file{ '/etc/graylog':
+    ensure => directory
+  }
+
+  file{ $ssl_config_dir:
+    ensure => directory
+  }
+
+$ssl_config_filename = 'openssl-graylog.cnf'
+$graylog_canonical_name = lookup('canonical_name')
+file { "${ssl_config_dir}/${ssl_config_filename}":
+    ensure  => present,
+    mode    => '0644',
+    owner   => 'root',
+    group   => 'root',
+    content => epp('profile/it/graylog_ssl_cfg.epp',
+      {
+      'country'           => lookup('country'),
+      'state'             => lookup('state'),
+      'locality'          => lookup('locality'),
+      'organization'      => lookup('organization'),
+      'division'          => lookup('division'),
+      'canonical_name'    => lookup('canonical_name'),
+      'server_ip'         => lookup('server_ip_address'),
+      'alternative_dns_1' => lookup('alternative_dns_1'),
+      }
+    ),
+    require => [File[$ssl_config_dir], File_line["Update Graylog's JAVA_OPTS"]],
+  }
+
+$certificate_duration = lookup('certificate_duration')
+$ssl_key_algorithm = lookup('ssl_key_algorithm')
+$ssl_keyout_filename = 'pkcs5-plain.pem'
+$ssl_graylog_cert_filename = 'graylog-certificate.pem'
+exec{ 'Create SSL certificate':
+    path    => [ '/usr/bin', '/bin', '/usr/sbin' ],
+    command => "openssl req -x509 -days ${certificate_duration} -nodes -newkey ${ssl_key_algorithm} \
+      -config ${ssl_config_dir}/${ssl_config_filename} -keyout ${ssl_config_dir}/${ssl_keyout_filename} \
+      -out ${ssl_config_dir}/${ssl_graylog_cert_filename}",
+    onlyif  => 'test ! -f /etc/graylog/server/graylog-certificate.pem',
+    require => [File["${ssl_config_dir}/${ssl_config_filename}"]]
+}
+
+# Convert PKCS5 into PKCS8 encrypted
+
+$ssl_pkcs8_encrypted_filename = 'pkcs8-encrypted.pem'
```

Comment:
The PR just merged from @jhoblitt includes the letsencrypt module, which would avoid the need of creating "by hand" the self-sign certificates and then convert them to pkcs8